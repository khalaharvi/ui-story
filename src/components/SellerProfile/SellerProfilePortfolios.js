import React from 'react';
import PropTypes from 'prop-types';

import RelatedModuleTemplate from 'templates/RelatedModule';
import SellerProfilePortfolio from './SellerProfilePortfolio';
import { cx } from './SellerProfile';

const SellerProfilePortfolios = ({ portfolios, portfolioComponent: PortfolioComponent }) => (
  <RelatedModuleTemplate className={cx('portfolios')} title="Featured Portfolio">
    <div className={cx('portfolios-grid')}>
      {
        portfolios.map(portfolio => (
          <div key={portfolio.id} className={cx('portfolios-grid-item')}>
            <PortfolioComponent portfolio={portfolio} />
          </div>
        ))
      }
    </div>
  </RelatedModuleTemplate>
);

SellerProfilePortfolios.propTypes = {
  portfolioComponent: PropTypes.func,
  portfolios: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    image: PropTypes.string.isRequired,
    caption: PropTypes.string.isRequired
  })).isRequired
};

SellerProfilePortfolios.defaultProps = {
  portfolioComponent: props => <SellerProfilePortfolio {...props} />
};

export default SellerProfilePortfolios;
