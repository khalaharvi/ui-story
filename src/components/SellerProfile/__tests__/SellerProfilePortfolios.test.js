import React from 'react';
import { shallow } from 'enzyme';

import SellerProfilePortfolios from '../SellerProfilePortfolios';

const portfolios = [
  {
    id: 1,
    image: 'https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.family.room.1501108236.5191004.jpg?auto=format&fit=clip&q=60&w=670',
    caption: 'Caption'
  },
  {
    id: 2,
    image: 'https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.family.room.1501108239.2473252.jpg?auto=format&fit=clip&q=60&w=670',
    caption: 'Caption'
  },
  {
    id: 3,
    image: 'https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.outdoor.room.1491576804.5543432.jpg?auto=format&fit=clip&q=60&w=670',
    caption: 'Caption'
  },
  {
    id: 4,
    image: 'https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.kitchen.1491576661.6619911.jpg?auto=format&fit=clip&q=60&w=670',
    caption: 'Caption'
  },
  {
    id: 5,
    image: 'https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.living.1491576648.5075328.jpg?auto=format&fit=clip&q=60&w=670',
    caption: 'Caption'
  },
  {
    id: 6,
    image: 'https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.foyer.1501108239.790634.jpg?auto=format&fit=clip&q=60&w=670',
    caption: 'Caption'
  },
  {
    id: 7,
    image: 'https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.outdoor.room.1491576575.5107028.jpg?auto=format&fit=clip&q=60&w=670',
    caption: 'Caption'
  }
];

describe('<SellerProfilePortfolios />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <SellerProfilePortfolios portfolios={portfolios} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with portfolioComponent', () => {
    const wrapper = shallow(
      <SellerProfilePortfolios
        portfolios={portfolios}
        portfolioComponent={({ portfolio }) => <p>{portfolio.caption}</p>}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
