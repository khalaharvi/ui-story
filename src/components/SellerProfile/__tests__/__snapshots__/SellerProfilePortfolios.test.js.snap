// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SellerProfilePortfolios /> renders correctly 1`] = `
<RelatedModuleTemplate
  className="dh-seller-profile-portfolios"
  title="Featured Portfolio"
>
  <div
    className="dh-seller-profile-portfolios-grid"
  >
    <div
      className="dh-seller-profile-portfolios-grid-item"
      key="1"
    >
      <portfolioComponent
        portfolio={
          Object {
            "caption": "Caption",
            "id": 1,
            "image": "https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.family.room.1501108236.5191004.jpg?auto=format&fit=clip&q=60&w=670",
          }
        }
      />
    </div>
    <div
      className="dh-seller-profile-portfolios-grid-item"
      key="2"
    >
      <portfolioComponent
        portfolio={
          Object {
            "caption": "Caption",
            "id": 2,
            "image": "https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.family.room.1501108239.2473252.jpg?auto=format&fit=clip&q=60&w=670",
          }
        }
      />
    </div>
    <div
      className="dh-seller-profile-portfolios-grid-item"
      key="3"
    >
      <portfolioComponent
        portfolio={
          Object {
            "caption": "Caption",
            "id": 3,
            "image": "https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.outdoor.room.1491576804.5543432.jpg?auto=format&fit=clip&q=60&w=670",
          }
        }
      />
    </div>
    <div
      className="dh-seller-profile-portfolios-grid-item"
      key="4"
    >
      <portfolioComponent
        portfolio={
          Object {
            "caption": "Caption",
            "id": 4,
            "image": "https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.kitchen.1491576661.6619911.jpg?auto=format&fit=clip&q=60&w=670",
          }
        }
      />
    </div>
    <div
      className="dh-seller-profile-portfolios-grid-item"
      key="5"
    >
      <portfolioComponent
        portfolio={
          Object {
            "caption": "Caption",
            "id": 5,
            "image": "https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.living.1491576648.5075328.jpg?auto=format&fit=clip&q=60&w=670",
          }
        }
      />
    </div>
    <div
      className="dh-seller-profile-portfolios-grid-item"
      key="6"
    >
      <portfolioComponent
        portfolio={
          Object {
            "caption": "Caption",
            "id": 6,
            "image": "https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.foyer.1501108239.790634.jpg?auto=format&fit=clip&q=60&w=670",
          }
        }
      />
    </div>
    <div
      className="dh-seller-profile-portfolios-grid-item"
      key="7"
    >
      <portfolioComponent
        portfolio={
          Object {
            "caption": "Caption",
            "id": 7,
            "image": "https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.outdoor.room.1491576575.5107028.jpg?auto=format&fit=clip&q=60&w=670",
          }
        }
      />
    </div>
  </div>
</RelatedModuleTemplate>
`;

exports[`<SellerProfilePortfolios /> renders correctly with portfolioComponent 1`] = `
<RelatedModuleTemplate
  className="dh-seller-profile-portfolios"
  title="Featured Portfolio"
>
  <div
    className="dh-seller-profile-portfolios-grid"
  >
    <div
      className="dh-seller-profile-portfolios-grid-item"
      key="1"
    >
      <portfolioComponent
        portfolio={
          Object {
            "caption": "Caption",
            "id": 1,
            "image": "https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.family.room.1501108236.5191004.jpg?auto=format&fit=clip&q=60&w=670",
          }
        }
      />
    </div>
    <div
      className="dh-seller-profile-portfolios-grid-item"
      key="2"
    >
      <portfolioComponent
        portfolio={
          Object {
            "caption": "Caption",
            "id": 2,
            "image": "https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.family.room.1501108239.2473252.jpg?auto=format&fit=clip&q=60&w=670",
          }
        }
      />
    </div>
    <div
      className="dh-seller-profile-portfolios-grid-item"
      key="3"
    >
      <portfolioComponent
        portfolio={
          Object {
            "caption": "Caption",
            "id": 3,
            "image": "https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.outdoor.room.1491576804.5543432.jpg?auto=format&fit=clip&q=60&w=670",
          }
        }
      />
    </div>
    <div
      className="dh-seller-profile-portfolios-grid-item"
      key="4"
    >
      <portfolioComponent
        portfolio={
          Object {
            "caption": "Caption",
            "id": 4,
            "image": "https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.kitchen.1491576661.6619911.jpg?auto=format&fit=clip&q=60&w=670",
          }
        }
      />
    </div>
    <div
      className="dh-seller-profile-portfolios-grid-item"
      key="5"
    >
      <portfolioComponent
        portfolio={
          Object {
            "caption": "Caption",
            "id": 5,
            "image": "https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.living.1491576648.5075328.jpg?auto=format&fit=clip&q=60&w=670",
          }
        }
      />
    </div>
    <div
      className="dh-seller-profile-portfolios-grid-item"
      key="6"
    >
      <portfolioComponent
        portfolio={
          Object {
            "caption": "Caption",
            "id": 6,
            "image": "https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.foyer.1501108239.790634.jpg?auto=format&fit=clip&q=60&w=670",
          }
        }
      />
    </div>
    <div
      className="dh-seller-profile-portfolios-grid-item"
      key="7"
    >
      <portfolioComponent
        portfolio={
          Object {
            "caption": "Caption",
            "id": 7,
            "image": "https://cdn.deringhall.com/portfolio_image/image/heather.wells.inc.portfolio.interiors.outdoor.room.1491576575.5107028.jpg?auto=format&fit=clip&q=60&w=670",
          }
        }
      />
    </div>
  </div>
</RelatedModuleTemplate>
`;
